resolwe/flow/executors/init_container.py

```diff
@@ -86,23 +105,27 @@ async def transfer_inputs(communicator: BaseCommunicator):
 
 
 async def download_to_location(
-    base_url: str,
-    files: dict,
+    files: list[dict[str, str]],
     from_connector: BaseStorageConnector,
     to_connector: BaseStorageConnector,
+    max_threads: int = 5,
 ):
     """Download missing paths.
 
     :raises DataTransferError: on failure.
     """
-    try:
-        logger.debug(f"Transfering: {from_connector} --> {to_connector}.")
-        t = Transfer(from_connector, to_connector)
-        t.transfer_objects(base_url, files)
-    except DataTransferError:
-        logger.exception(
-            "Data transfer error downloading data to base url {}.".format(base_url)
+    logger.info(f"Transfering data {from_connector} --> {to_connector}.")
+    transfer = Transfer(from_connector, to_connector)
+    # Start futures and evaluate their results. If exception occured it will
+    # be re-raised.
+    (
```

Comment:
I don't like list comprehension here, as it only makes code longer and more complex.

Reply:
OK.